### PR TITLE
Python 3 compatibility, allow preprocess_spec.py to be imported, and add descriptions to make Swagger validate

### PR DESCRIPTION
**Pull request by ceridwen**
This is straightforward patch to address #19.  It also makes three more changes to address my use-case:
1. Encloses `sys.exit(main))` in `if __name__ == '__main__'` to allow access through Python's import functionality as well as through a subprocess call.
2. Adds the `__future__` import for the print function and turns print statements into print functions.
3. Replaces `iteritems()` with `items()`.

The first two of these shouldn't change anything but allow the code to be imported and to run on Python 3.  The third has possible performance implications because on Python 2 `items()` creates a list rather than an iterator.  The standard way of working around this difference is to use [six](https://pythonhosted.org/six/), but if you don't want to bother with the complication, just tell me and I'll remove that change from the pull request.

### CODE REVIEW

File: openapi/preprocess_spec.py [hunk 1]
```diff
@@ -121,7 +123,7 @@ def process_swagger(spec):
         apply_func_to_spec_operations(
             spec, remove_watch_operations, operation_ids)
     except PreprocessingException as e:
-        print(e.message)
```
Comment (brendandburns) [inline openapi/preprocess_spec.py]: why is this needed?

**Comment by brendandburns** [comment]
Generally LGTM, one small question.

You need to sign the cla.

**Comment by mbohlool** [comment]
Nice PR. Thanks.

Please note that preprocess_spec.py is now running inside container and I think it is running with python 2.7. The change looks compatible with 2.7 but have you tried running generator (that creates and run inside container) to make sure everything works?

**Comment by ceridwen** [comment]
I have signed up as an employee, but it's still complaining that I don't have a CLA.  I honestly don't have any more time to waste on struggling with this sign up process, so unless there's a straightforward fix...

**Comment by mbohlool** [comment]
The fix was you commenting again so the bot check your signature :)

Just one last request, can you run the generation (that needs docker) to make sure your changes didn't break that. I can do that, but will take a little longer for me to do so. If you don't mind this sitting around for a little while, ignore my request :)

File: openapi/preprocess_spec.py (hunk 1 above)
Comment (ceridwen) [inline openapi/preprocess_spec.py]: It isn't, not sure why 2to3 added that.  I removed it.

**Comment by ceridwen** [comment]
I tried but gave up when I had to manually edit Maven configuration files to get the generation script to run.  You'll have to do it, sorry.

**Comment by mbohlool** [comment]
Please report back the problem so I can fix it. The point of containerizing the generation was to make it easier for anybody to run it. A change to maven configuration should not be necessary.

I will run the generation and confirm this is working but I would really appreciate it if you report back (here or in an issue) why you needed to change maven configuration to make it work.

File: openapi/preprocess_spec.py [hunk 2]
```diff
@@ -205,23 +207,23 @@ def find_replace_ref_recursive(root, ref_name, replace_map):
...
             find_replace_ref_recursive(v, ref_name, replace_map)
 
 
 def inline_primitive_models(spec):
     to_remove_models = []
-    for k, v in spec['definitions'].iteritems():
+    for k, v in spec['definitions'].items():
         if "properties" not in v:
-            print("Making primitive mode `%s` inline ..." % k)
+            print(("Making primitive mode `%s` inline ..." % k))
```
Comment (mbohlool) [inline openapi/preprocess_spec.py]: You have another double parentheses here :) Please squash all of your commits into one after this fix.

**Comment by ceridwen** [comment]
I brought the repository up to date, got all the remaining superfluous parentheses (I hope), and squashed the commits.  My mistake, sorry, about building it---I was running `generate_client_in_container.sh`, when the actual entrypoint is just `docker build`?  `docker build` finished with:

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] swagger-codegen-project ............................ SUCCESS [ 18.759 s]
[INFO] swagger-codegen (core library) ..................... SUCCESS [01:08 min]
[INFO] swagger-codegen (maven-plugin) ..................... SUCCESS [ 28.401 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:59 min
[INFO] Finished at: 2017-08-24T18:33:44Z
[INFO] Final Memory: 51M/284M
[INFO] ------------------------------------------------------------------------
...
Successfully built ace032e990dd
```

I was expecting a shell script that invoked `docker build` and copied the resulting generated code back to my computer.  You might add a note to that effect to prevent other people from confusing themselves the way I did.

**Comment by mbohlool** [comment]
This repo's README file explained how to run the generator. Simply you can go to one of the language repos (java, javascript, typescript), clone them, go to their root and run THIS_REPO/openapi/autoupdate.sh on it (or more specifically $lang out_dir setting_file).

Anyway I ran this on python client and it worked fine. So it should be good to go. Thanks.
